erpnext/hr/doctype/expense_claim/expense_claim.py

```diff
@@ -360,3 +361,43 @@ def get_expense_claim(
 	)
 
 	return expense_claim
+
+# api to create expense claim
+@frappe.whitelist()
+def create_expense_claim(subject, email, expense_date, expense_type, amount):
+	# fetch employee code from email
+	employee = frappe.get_all("Employee", or_filters={
+		"prefered_email": email,
+		"company_email": email,
+		"personal_email": email
+	})
+
+	if employee and employee[0]:
+		employee = frappe.get_doc("Employee", employee[0].name)
+	else:
+		frappe.throw(_("Employee not found."))
+
+	if employee.department:
+		department = frappe.get_doc("Department", employee.department)
+		if department.expense_approvers:
+			expense_approver = department.expense_approvers[ os.urandom(1)[0] % len(department.expense_approvers)].approver
+		else:
+			frappe.throw(_("Expense Approver not found for {0}").format(email))
```

Comment:
don't do a throw

Reply:
@neilLasrado who is going to approve the request?